File: _includes/code/csharp/ReferenceCollectionsTest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Weaviate.Client;
using Weaviate.Client.Models;
using Xunit;
using static Weaviate.Client.Models.VectorIndexConfig;

// Reference: Collection definition - the examples rendered on
// docs/weaviate/config-refs/collections.mdx. They mirror the Python examples
// in _includes/code/config-refs/reference.collections.py one-for-one.

// This attribute ensures that tests in this class do not run in parallel,
// which is important because they share a client and perform cleanup operations
// that could otherwise interfere with each other.
[Collection("Sequential")]
public class ReferenceCollectionsTest : IAsyncLifetime
{
    private static readonly WeaviateClient client;

    // Static constructor acts like JUnit's @BeforeAll
    static ReferenceCollectionsTest()
    {
        // Instantiate the client with the OpenAI API key
        string openaiApiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY");
        if (string.IsNullOrWhiteSpace(openaiApiKey))
        {
            throw new ArgumentException("Please set the OPENAI_API_KEY environment variable.");
        }

        var headers = new Dictionary<string, string> { { "X-OpenAI-Api-Key", openaiApiKey } };
        client = Connect
            .Local(hostname: "localhost", restPort: 8080, headers: headers)
            .GetAwaiter()
            .GetResult();
    }

    // InitializeAsync is called before each test. We ensure all collections are deleted.
    public async Task InitializeAsync()
    {
        await client.Collections.DeleteAll();
    }

    // DisposeAsync acts like JUnit's @AfterEach, cleaning up after every test.
    public Task DisposeAsync()
    {
        // No action needed here since cleanup happens in InitializeAsync before the next test.
        return Task.CompletedTask;
    }

    [Fact]
    public async Task TestReferenceFullDefinition()
    {
        // START ReferenceFullDefinition
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "Article",
                Description = "A collection of articles",
                Properties = [Property.Text("title"), Property.Text("body")],
                VectorConfig = new VectorConfigList
                {
                    Configure.Vector(
                        "default",
                        v => v.Text2VecTransformers(),
                        sourceProperties: ["title", "body"],
                        index: new VectorIndex.HNSW()
                        {
                            EfConstruction = 300,
                            Distance = VectorDistance.Cosine,
                            FilterStrategy = VectorIndexFilterStrategy.Sweeping,
                        }
                    ),
                },
                MultiTenancyConfig = new MultiTenancyConfig { Enabled = false },
                ShardingConfig = new ShardingConfig
                {
                    VirtualPerPhysical = 128,
                    DesiredCount = 1,
                    DesiredVirtualCount = 128,
                },
                ReplicationConfig = new ReplicationConfig
                {
                    Factor = 1,
                    DeletionStrategy = DeletionStrategy.TimeBasedResolution,
                },
            }
        );
        // END ReferenceFullDefinition

        var config = await client.Collections.Export("Article");
        Assert.Equal("A collection of articles", config.Description);
        Assert.Equal(2, config.Properties.Length);
        Assert.Equal(128, config.ShardingConfig.DesiredVirtualCount);
    }

    [Fact]
    public async Task TestReferenceVectorizer()
    {
        // START ReferenceVectorizer
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "Article",
                // highlight-start
                VectorConfig = new VectorConfigList
                {
                    Configure.Vector(
                        "default", // (Optional) Set the name of the vector, default name is "default"
                        v => v.Text2VecTransformers(vectorizeCollectionName: true),
                        sourceProperties: ["title", "body"], // (Optional) Set the source property(ies)
                        index: new VectorIndex.HNSW() // (Optional) Set vector index options
                        {
                            EfConstruction = 300,
                            Distance = VectorDistance.Cosine,
                            FilterStrategy = VectorIndexFilterStrategy.Sweeping,
                        }
                    ),
                },
                // highlight-end
                // properties configuration is optional
                Properties = [Property.Text("title"), Property.Text("body")],
            }
        );
        // END ReferenceVectorizer

        var config = await client.Collections.Export("Article");
        Assert.True(config.VectorConfig.ContainsKey("default"));
    }

    [Fact]
    public async Task TestReferenceNamedVectors()
    {
        // START ReferenceNamedVectors
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "Article",
                // highlight-start
                VectorConfig = new VectorConfigList
                {
                    Configure.Vector(
                        "default",
                        v => v.Text2VecTransformers(vectorizeCollectionName: true),
                        sourceProperties: ["title", "body"],
                        index: new VectorIndex.HNSW()
                        {
                            EfConstruction = 300,
                            Distance = VectorDistance.Cosine,
                            FilterStrategy = VectorIndexFilterStrategy.Sweeping,
                        }
                    ),
                    Configure.Vector(
                        "body_vectors",
                        v => v.Text2VecTransformers(),
                        sourceProperties: ["body"],
                        index: new VectorIndex.Flat()
                    ),
                },
                // highlight-end
                // properties configuration is optional
                Properties = [Property.Text("title"), Property.Text("body")],
            }
        );
        // END ReferenceNamedVectors

        var config = await client.Collections.Export("Article");
        Assert.Equal(2, config.VectorConfig.Count);
        Assert.Contains("default", config.VectorConfig.Keys);
        Assert.Contains("body_vectors", config.VectorConfig.Keys);
    }

    [Fact]
    public async Task TestReferencePropertyOptions()
    {
        // START ReferencePropertyOptions
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "Article",
                VectorConfig = new VectorConfigList
                {
                    Configure.Vector("default", v => v.Text2VecTransformers()),
                },
                // highlight-start
                Properties =
                [
                    Property.Text(
                        "title",
                        description: "The article headline",
                        tokenization: PropertyTokenization.Word // How the text is split for the inverted index
                    ),
                    Property.Text("description", indexSearchable: false), // Exclude it from keyword search
                    Property.Number("rating", indexRangeFilters: true), // Index for range-based filtering
                ],
                // highlight-end
            }
        );
        // END ReferencePropertyOptions

        var config = await client.Collections.Export("Article");
        Assert.Equal(3, config.Properties.Length);
    }

    [Fact]
    public async Task TestReferenceReplicationSettings()
    {
        // START ReferenceReplicationSettings
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "Article",
                // highlight-start
                ReplicationConfig = new ReplicationConfig
                {
                    Factor = 1,
                    DeletionStrategy = DeletionStrategy.TimeBasedResolution,
                    AsyncEnabled = true,
                    AsyncConfig = new ReplicationAsyncConfig
                    {
                        HashtreeHeight = 16,
                        Frequency = 30,
                    },
                },
                // highlight-end
            }
        );
        // END ReferenceReplicationSettings

        var config = await client.Collections.Export("Article");
        Assert.Equal(
            DeletionStrategy.TimeBasedResolution,
            config.ReplicationConfig.DeletionStrategy
        );
    }
}

File: _includes/code/java-v6/src/test/java/ReferenceCollectionsTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.Replication.DeletionStrategy;
import io.weaviate.client6.v1.api.collections.Tokenization;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import io.weaviate.client6.v1.api.collections.vectorindex.Distance;
import io.weaviate.client6.v1.api.collections.vectorindex.Flat;
import io.weaviate.client6.v1.api.collections.vectorindex.Hnsw;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Map;
import static org.assertj.core.api.Assertions.assertThat;

// Reference: Collection definition - the examples rendered on
// docs/weaviate/config-refs/collections.mdx. They mirror the Python examples
// in _includes/code/config-refs/reference.collections.py one-for-one.
class ReferenceCollectionsTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() throws IOException {
    // Instantiate the client with the OpenAI API key
    String openaiApiKey = System.getenv("OPENAI_API_KEY");
    assertThat(openaiApiKey).isNotBlank()
        .withFailMessage("Please set the OPENAI_API_KEY environment variable.");

    client = WeaviateClient.connectToLocal(
        config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));
    client.collections.deleteAll();
  }

  @AfterEach
  public void afterEach() throws IOException {
    // Clean up all collections after each test
    client.collections.deleteAll();
  }

  @Test
  void testReferenceFullDefinition() throws IOException {
    // START ReferenceFullDefinition
    client.collections.create("Article",
        col -> col
            .description("A collection of articles")
            .properties(Property.text("title"), Property.text("body"))
            .vectorConfig(VectorConfig.text2vecTransformers("default",
                vec -> vec.sourceProperties("title", "body")
                    .vectorIndex(Hnsw.of(hnsw -> hnsw.efConstruction(300)
                        .distance(Distance.COSINE)
                        .filterStrategy(Hnsw.FilterStrategy.SWEEPING)))))
            .multiTenancy(mt -> mt.enabled(false))
            .sharding(s -> s.virtualPerPhysical(128)
                .desiredCount(1)
                .desiredVirtualCount(128))
            .replication(rep -> rep.replicationFactor(1)
                .deletionStrategy(DeletionStrategy.TIME_BASED_RESOLUTION)));
    // END ReferenceFullDefinition

    var config = client.collections.getConfig("Article").get();
    assertThat(config.description()).isEqualTo("A collection of articles");
    assertThat(config.properties()).hasSize(2);
    assertThat(config.vectors()).containsKey("default");
    assertThat(config.sharding().desiredVirtualCount()).isEqualTo(128);
  }

  @Test
  void testReferenceVectorizer() throws IOException {
    // START ReferenceVectorizer
    client.collections.create("Article",
        col -> col
            // highlight-start
            .vectorConfig(VectorConfig.text2vecTransformers("default", // (Optional) Set the name of the vector, default name is "default"
                vec -> vec.sourceProperties("title", "body") // (Optional) Set the source property(ies)
                    .vectorIndex(Hnsw.of(hnsw -> hnsw.efConstruction(300)
                        .distance(Distance.COSINE)
                        .filterStrategy(Hnsw.FilterStrategy.SWEEPING))))) // (Optional) Set vector index options
            // highlight-end
            .properties( // properties configuration is optional
                Property.text("title", p -> p.vectorizePropertyName(true)),
                Property.text("body")));
    // END ReferenceVectorizer

    var config = client.collections.getConfig("Article").get();
    assertThat(config.vectors()).containsKey("default");
  }

  @Test
  void testReferenceNamedVectors() throws IOException {
    // START ReferenceNamedVectors
    client.collections.create("Article",
        col -> col
            // highlight-start
            .vectorConfig(
                VectorConfig.text2vecTransformers("default",
                    vec -> vec.sourceProperties("title", "body")
                        .vectorIndex(Hnsw.of(hnsw -> hnsw.efConstruction(300)
                            .distance(Distance.COSINE)
                            .filterStrategy(Hnsw.FilterStrategy.SWEEPING)))),
                VectorConfig.text2vecTransformers("body_vectors",
                    vec -> vec.sourceProperties("body")
                        .vectorIndex(Flat.of())))
            // highlight-end
            .properties( // properties configuration is optional
                Property.text("title", p -> p.vectorizePropertyName(true)),
                Property.text("body")));
    // END ReferenceNamedVectors

    var config = client.collections.getConfig("Article").get();
    assertThat(config.vectors()).hasSize(2).containsKeys("default", "body_vectors");
  }

  @Test
  void testReferencePropertyOptions() throws IOException {
    // START ReferencePropertyOptions
    client.collections.create("Article",
        col -> col
            .vectorConfig(VectorConfig.text2vecTransformers())
            .properties(
                Property.text("title",
                    p -> p.description("The article headline")
                        .tokenization(Tokenization.WORD)), // How the text is split for the inverted index
                Property.text("description",
                    p -> p.skipVectorization(true) // Exclude this property from the vector
                        .indexSearchable(false)), // Exclude it from keyword search
                Property.number("rating",
                    p -> p.indexRangeFilters(true)))); // Index for range-based filtering
    // END ReferencePropertyOptions

    var config = client.collections.getConfig("Article").get();
    assertThat(config.properties()).hasSize(3);
  }
}
